### PR TITLE
[10.0] [FIX] stock: Fix product template compute quantities depends

### PR DESCRIPTION
**Pull request by FFernandez-PlanetaTIC**
**Description of the issue/feature this PR addresses:**

The absence of `@api.depends` in method `_compute_quantities` of model `product.template` is causing a glitch:

https://github.com/OCA/connector-prestashop/pull/101#issuecomment-448229084

**Current behavior before PR:**

The `qty_available` of a `product.template` is **not** recomputed when it is done for one of its `product.product`:

```
(virtualenv-10.0) paco@coloma:~/workspace/PlanetaTIC/odoo$ python odoo-bin shell -d 10.0-fix-stock-product_template_compute_quantities_depends
...
>>> product = env['product.product'].create({'name': 'Test', 'type': 'product'})
>>> product.qty_available
0.0
>>> product.product_tmpl_id.qty_available
0.0
>>> 
>>> warehouse = env['stock.warehouse'].search([], limit=1)
>>> 
>>> qty_change = env['stock.change.product.qty'].create({'location_id': warehouse.lot_stock_id.id, 'product_id': product.id, 'new_quantity': 10})
>>> qty_change.change_product_qty()
{'type': 'ir.actions.act_window_close'}
>>> 
>>> product.qty_available
10.0
>>> product.product_tmpl_id.qty_available
0.0
>>> 
>>> env.clear()
>>> product.product_tmpl_id.qty_available
10.0
```

**Desired behavior after PR is merged:**

The `qty_available` of a `product.template` is recomputed when it is done for one of its `product.product`:

```
(virtualenv-10.0) paco@coloma:~/workspace/PlanetaTIC/odoo$ python odoo-bin shell -d 10.0-fix-stock-product_template_compute_quantities_depends
...
>>> product = env['product.product'].create({'name': 'Test', 'type': 'product'})
>>> product.qty_available
0.0
>>> product.product_tmpl_id.qty_available
0.0
>>> 
>>> warehouse = env['stock.warehouse'].search([], limit=1)
>>> 
>>> qty_change = env['stock.change.product.qty'].create({'location_id': warehouse.lot_stock_id.id, 'product_id': product.id, 'new_quantity': 10})
>>> qty_change.change_product_qty()
{'type': 'ir.actions.act_window_close'}
>>> 
>>> product.qty_available
10.0
>>> product.product_tmpl_id.qty_available
10.0

```

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@jco-odoo can you check this one?

**Comment by pedrobaeza** [comment]
@mart-e the number of moves indeed is high, but when you receive stock, you have to recompute it. At the end, it's going to impact the same as invalidating cache on that step, so I think it's more correct to do this for consistency.

**Comment by jco-odoo** [comment]
I tend to agree with @mart-e that the inconsistency does not outweigh the performance loss with the fix.  Anyone perf numbers?

**Comment by pedrobaeza** [comment]
Currently the facts are:

* The available quantity is not correct in all cases depending on the state of the cache.
* In the worst case, if you want to assure the correct quantity, you have to invalidate the cache, and this is worst than recompute it when needed, as you invalidate whole cache. There are not going to be extra recomputations, only the needed.

Are we seriously preferring to add a lot of cache invalidations than this?

File: addons/stock/models/product.py [hunk 1]
```diff
@@ -432,6 +432,11 @@ class ProductTemplate(models.Model):
         relation="stock.location.route", string="Category Routes",
         related='categ_id.total_route_ids')
 
+    @api.depends(
+        'product_variant_ids',
+        'product_variant_ids.stock_quant_ids',
+        'product_variant_ids.stock_move_ids',
```
Comment (pedrobaeza) [inline addons/stock/models/product.py]: I think this depends line is not needed at all

**Comment by mart-e** [comment]
Actually, as the field is not stored, it is probably not that bad. You will have an impact on the performance in a transaction where you both change stock move and access the quantity but if you only access one of the two, it should have no impact.

**Comment by pedrobaeza** [comment]
@mart-e then this can be accepted?

File: addons/stock/models/product.py (hunk 1 above)
Comment (FFernandez-PlanetaTIC) [reply addons/stock/models/product.py]: Removed

**Comment by mart-e** [comment]
> then this can be accepted?

Yes it can, thanks for the PR :+1: 

@robodoo r+

**Comment by pedrobaeza** [comment]
Thanks!

**Comment by rdeodoo** [comment]
Looks like this one got stucked in r+ mode?

**Comment by pedrobaeza** [comment]
Oh, you're right! I hope it can finally arrive.

**Comment by mart-e** [comment]
@robodoo r+ ? :thinking:

**Comment by mart-e** [comment]
@robodoo I am getting tired... retry !

**Pull request closed without merging**